consensus/consensus-types/src/common.rs

```diff
@@ -278,30 +278,42 @@ impl Payload {
         }
     }
 
+    fn verify_proofs(
+        proofs: &[ProofOfStore],
+        validator: &ValidatorVerifier,
+        proof_cache: &ProofCache,
+    ) -> anyhow::Result<()> {
+        let unverified: Vec<_> = proofs
+            .iter()
+            .filter(|proof| {
+                proof_cache.get(proof.info()).map_or(true, |cached_proof| {
+                    cached_proof != *proof.multi_signature()
+                })
+            })
+            .collect();
+        unverified
+            .par_iter()
+            .with_min_len(2)
```

Comment:
minimum length of 2 seems low as the overhead of spawning a rayon task is pretty high. How about something like 4 or 8?

Reply:
8 gives us around 16ms delay when there are cache misses. I didn't notice much overhead with 2 (4 ms delay). Are you more concerned about how the overhead affects other parts of the system?

Reply:
As discussed offline, given each proof verification takes 2 ms on a single thread, the overhead of spawning a rayon task should be negligible as compared to the task itself.  So this looks good to me.